packages/block-editor/src/layouts/grid.js

```diff
@@ -290,18 +307,32 @@ function GridLayoutColumnsAndRowsControl( {
 						<NumberControl
 							size="__unstable-large"
 							onChange={ ( value ) => {
-								/**
-								 * If the input is cleared, avoid switching
-								 * back to "Auto" by setting a value of "1".
-								 */
-								const validValue = value !== '' ? value : '1';
-								onChange( {
-									...layout,
-									columnCount:
-										window.__experimentalEnableGridInteractivity
-											? parseInt( value, 10 ) || null
-											: parseInt( validValue, 10 ),
-								} );
+								if (
+									window.__experimentalEnableGridInteractivity
+								) {
+									// Allow unsetting the column count when in auto mode.
+									const defaultColumnCount = isManualPlacement
+										? 1
+										: null; // TODO: Why not undefined?
```

Comment:
Need to look into this. Shouldn't we remove the attribute instead of making it `null`?

Reply:
Yeah it should be unsettable here, we're no longer using columnCount to distinguish Auto from Manual mode so it shouldn't matter if it exists or not.

Reply:
Yep makes sense. I'll make it remove the attribute instead of setting it to null, though.